pkg/resources/resource_monitor.go

```diff
@@ -2,504 +2,450 @@ package resources
+			"suspend_trigger",
+			"suspend_immediate_trigger",
+		}); err != nil {
+			return diag.FromErr(err)
 		}
-	} else {
-		if err := d.Set("suspend_immediate_trigger", nil); err != nil {
-			return err
+
+		if err = d.Set(ShowOutputAttributeName, []map[string]any{schemas.ResourceMonitorToSchema(resourceMonitor)}); err != nil {
+			return diag.FromErr(err)
 		}
-	}
 
-	if err := d.Set("notify_triggers", resourceMonitor.NotifyTriggers); err != nil {
-		return err
-	}
+		if err := d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()); err != nil {
+			return diag.FromErr(err)
+		}
 
-	// Account level
-	if err := d.Set("set_for_account", resourceMonitor.Level == sdk.ResourceMonitorLevelAccount); err != nil {
-		return err
+		return nil
 	}
-
-	return err
 }
 
-// UpdateResourceMonitor implements schema.UpdateFunc.
-func UpdateResourceMonitor(d *schema.ResourceData, meta interface{}) error {
+func UpdateResourceMonitor(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	name := d.Get("name").(string)
 
-	check := checkAccountAgainstWarehouses(d, name)
-
-	if check != nil {
-		return check
+	id, err := sdk.ParseAccountObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
 	}
 
-	objectIdentifier := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
-
-	ctx := context.Background()
 	var runSetStatement bool
-
+	var runUnsetStatement bool
 	opts := sdk.AlterResourceMonitorOptions{}
-
 	set := sdk.ResourceMonitorSet{}
+	unset := sdk.ResourceMonitorUnset{}
+
 	if d.HasChange("credit_quota") {
 		runSetStatement = true
-		set.CreditQuota = sdk.Pointer(d.Get("credit_quota").(int))
+		if v, ok := d.GetOk("credit_quota"); ok {
+			set.CreditQuota = sdk.Pointer(v.(int))
+		} else {
+			runUnsetStatement = true
+			unset.CreditQuota = sdk.Bool(true)
+		}
 	}
 
-	if d.HasChange("frequency") || d.HasChange("start_timestamp") {
+	if (d.HasChange("frequency") || d.HasChange("start_timestamp")) &&
+		(d.Get("frequency").(string) != "" && d.Get("start_timestamp").(string) != "") {
 		runSetStatement = true
-		frequency, err := sdk.FrequencyFromString(d.Get("frequency").(string))
+		frequency, err := sdk.ToResourceMonitorFrequency(d.Get("frequency").(string))
 		if err != nil {
-			return err
+			return diag.FromErr(err)
 		}
 		set.Frequency = frequency
 		set.StartTimestamp = sdk.Pointer(d.Get("start_timestamp").(string))
 	}
 
 	if d.HasChange("end_timestamp") {
-		runSetStatement = true
-		set.EndTimestamp = sdk.Pointer(d.Get("end_timestamp").(string))
-	}
-
-	if d.HasChange("notify_users") {
-		runSetStatement = true
-
-		userNames := expandStringList(d.Get("notify_users").(*schema.Set).List())
-		users := []sdk.NotifiedUser{}
-		for _, name := range userNames {
-			users = append(users, sdk.NotifiedUser{Name: name})
-		}
-		set.NotifyUsers = &sdk.NotifyUsers{
-			Users: users,
+		if v, ok := d.GetOk("end_timestamp"); ok {
+			runSetStatement = true
```

Comment:
runSetStatement and runUnsetStatement are left on purpose? (we are not using them any more in the reworked resources)

Reply:
True, I removed them